examples/fluids/problems/newtonian.c

```diff
@@ -363,8 +363,10 @@ PetscErrorCode NS_NEWTONIAN_IG(ProblemData *problem, DM dm, void *ctx, SimpleBC
   problem->apply_vol_rhs.qfunction_context = newtonian_ig_context;
   PetscCallCeed(ceed, CeedQFunctionContextReferenceCopy(newtonian_ig_context, &problem->apply_vol_ifunction.qfunction_context));
   PetscCallCeed(ceed, CeedQFunctionContextReferenceCopy(newtonian_ig_context, &problem->apply_vol_ijacobian.qfunction_context));
-  PetscCallCeed(ceed, CeedQFunctionContextReferenceCopy(newtonian_ig_context, &problem->apply_inflow.qfunction_context));
-  PetscCallCeed(ceed, CeedQFunctionContextReferenceCopy(newtonian_ig_context, &problem->apply_inflow_jacobian.qfunction_context));
+  if (bc->num_inflow > 0) {
```

Comment:
Its safe to Destroy NULL, so is there a way to get around needing so many > 0 checks?

Reply:
This sort of reduction in special casing logic would probably be the sort of thing that could spread all over the place and would probably be worth being its own branch

Reply:
Yeah, I think so. I realized that the context destroying code in `navierstokes.c` repeated a few values, so I fixed that in an update. I'll double check if that absolves the need for the if statement here.

As for the other `> 0` checks, that necessary for the boundary condition setup in general. Something I want to get rid of with a re-write of the BC handling code. But that's for another day.

Reply:
fair, just wanted to put that on the radar since its bit me a lot in these projects